proposals/NNNN-align-VidepStreamingParameter-with-capability.md

```diff
@@ -0,0 +1,124 @@
+                if ([self.customEncoderSettings valueForKey:key] < self.videoEncoderSettings[key]) {
+                    self.videoEncoderSettings[key] = [self.customEncoderSettings valueForKey:key];
+                }
+            } else {
+                self.videoEncoderSettings[key] = [self.customEncoderSettings valueForKey:key];
+            }
+        }
+		...
+}
+```
+
+### Detailed design for Java Suite library
+
+Java Suite library updates custom `VideoStreamingParameters` when `VideoStreamingCapability` is given by head unit. `VideoStreamingParameter#update` method does the job.
+
+For Resolution and Scale, the `update()` method already takes care of capability as well as making some adjustment when `vehicleMake` is specified, like below:
+
+VideoStreamingParameter.java (**no change seems to be needed for Resolution and Scale**)
+
+```java
+public void update(VideoStreamingCapability capability, String vehicleMake) {
+	...
+    // For resolution and scale, the capability values should be taken than parameters specified by developers.
+    if (capability.getScale() != null) {
+        scale = capability.getScale();
+    }
+    ImageResolution resolution = capability.getPreferredResolution();
+    if (resolution != null) {
+
+        if (vehicleMake != null) {
+            if ((vehicleMake.contains("Ford") || vehicleMake.contains("Lincoln")) && ((resolution.getResolutionHeight() != null && resolution.getResolutionHeight() > 800) || (resolution.getResolutionWidth() != null && resolution.getResolutionWidth() > 800))) {
+                scale = 1.0 / 0.75;
+            }
+        }
+
+        if (resolution.getResolutionHeight() != null && resolution.getResolutionHeight() > 0) {
+            this.resolution.setResolutionHeight((int) (resolution.getResolutionHeight() / scale));
+        }
+        if (resolution.getResolutionWidth() != null && resolution.getResolutionWidth() > 0) {
+            this.resolution.setResolutionWidth((int) (resolution.getResolutionWidth() / scale));
+        }
+    }
+	....
+}
+```
+
+In the same method, however, bitrate and frame rate should take lower value approach, like below:
+
+VideoStreamingParameter.java (**code snippet that needs to be changed**)
+
+```java
+public void update(VideoStreamingCapability capability, String vehicleMake) {
+	...
+    if (capability.getMaxBitrate() != null) {
+        this.bitrate = Math.min(this.bitrate, capability.getMaxBitrate() * 1000);
+    } // NOTE: the unit of maxBitrate in getSystemCapability is kbps.
+	
+	...
+	if (capability.getPreferredFPS() != null) {
+        this.frameRate = Math.min(this.frameRate, capability.getPreferredFPS());
+    }
+    ...
+}
+```
+
+## Potential downsides
+
+There should be no downside by aligning bitrate and frame rate with `VideoStreamingCapability`.
+
+## Impact on existing code
+
+No impact on existing code, as there's no API change.
```

Comment:
As both iOS and Java Suite are listed as `Impacted platforms`, and contain changes within the `Proposed solution` section, please provide more information on what is being changed within those libraries for this proposal.

Reply:
Understood. I have added a possible impact. Thanks.